src/diracx/db/sandbox_metadata/db.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+import datetime
+
+import sqlalchemy
+
+from ..utils import BaseDB
+from .schema import Base as SandboxMetadataDBBase
+from .schema import sb_Owners, sb_SandBoxes
+
+
+class SandboxMetadataDB(BaseDB):
+    # This needs to be here for the BaseDB to create the engine
+    metadata = SandboxMetadataDBBase.metadata
+
+    async def _get_put_owner(self, owner: str, owner_group: str) -> int:
+        """adds a new owner/ownerGroup pairs, while returning their ID if already existing
+
+        Args:
+            owner (str): _description_
+            owner_group (str): _description_
+        """
+        stmt = sqlalchemy.select(sb_Owners.OwnerID).where(
+            sb_Owners.Owner == owner, sb_Owners.OwnerGroup == owner_group
+        )
+        result = await self.conn.execute(stmt)
+        res = result.fetchall()
+        if res:
+            return res[0][0]
```

Comment:
I think `scalar_one_or_none` is the idiomatic way of doing this:

```suggestion
        if owner_id := result.scalar_one_or_none():
            return owner_id
```

(with the benifit that if the impossible happens and multiple results are returned we get an exception)